src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -218,7 +218,9 @@ private function registerMetadataConfiguration(ContainerBuilder $container, arra
 
         $container->getDefinition('api_platform.metadata.extractor.xml')->addArgument($xmlResources);
 
-        if (class_exists(Annotation::class)) {
+        // The "annotation_reader" service is only available if DoctrineBundle is loaded.
+        $bundles = $container->getParameter('kernel.bundles');
```

Comment:
`$bundles` is already defined here: https://github.com/fgrandjean/core/blob/a6bb41d21f37898d80daae52336418e5310d4daf/src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php#L118

You can just pass it as parameter of all `register*` methods.